File: cli/src/commands/list.rs
use anyhow::Result;
use clap::Args as ClapArgs;
use std::path::PathBuf;

use crate::manifest::Manifest;
use crate::paths;

#[derive(ClapArgs)]
pub struct Args {
    /// Install target to inspect. Defaults to ~/.claude/skills/.
    #[arg(long, short = 't')]
    target: Option<PathBuf>,
}

pub fn run(args: Args) -> Result<()> {
    let target = paths::resolve_target(args.target)?;
    let manifest_path = paths::manifest_path(&target);

    if !manifest_path.exists() {
        println!("no Quaere skills installed at {}", target.display());
        println!("run `quaere install` to set them up");
        return Ok(());
    }

    let manifest = Manifest::read(&manifest_path)?;
    println!("target  {}", target.display());
    println!("version {}", manifest.quaere_version);
    if manifest.skills.is_empty() {
        println!("skills  (none recorded)");
    } else {
        println!("skills");
        for name in &manifest.skills {
            let present = target.join(name).is_dir();
            let marker = if present { " " } else { "!" };
            println!("  {} {}", marker, name);
        }
        let missing: Vec<_> = manifest
            .skills
            .iter()
            .filter(|n| !target.join(n).is_dir())
            .collect();
        if !missing.is_empty() {
            eprintln!(
                "warning: {} skill(s) in manifest are missing on disk; run `quaere doctor`",
                missing.len()
            );
        }
    }
    Ok(())
}

File: cli/src/commands/list.ts
import { existsSync } from 'node:fs'
import { readManifest } from '../lib/manifest.js'
import { getClaudeSkillsDir, getCodexSkillsDir } from '../lib/paths.js'
import { getCoreSkills, getExtensionSkills } from '../lib/skills.js'

interface TargetInfo {
  label: string
  dir: string
}

const TARGETS: TargetInfo[] = [
  { label: 'Claude Code', dir: getClaudeSkillsDir() },
  { label: 'Codex CLI', dir: getCodexSkillsDir() },
]

export function runList(): void {
  const coreNames = new Set(getCoreSkills().map((s) => s.name))
  const extensionNames = new Set(getExtensionSkills().map((s) => s.name))
  const kindOf = (name: string): string => {
    if (coreNames.has(name)) return 'core'
    if (extensionNames.has(name)) return 'extension'
    return 'other'
  }

  let found = false

  for (const { label, dir } of TARGETS) {
    const manifest = readManifest(dir)
    if (!manifest) continue

    found = true
    console.log(`\n${label}  (${dir})`)
    console.log(`  version  ${manifest.quaere_version}`)

    if (manifest.skills.length === 0) {
      console.log('  (no skills recorded)')
      continue
    }

    for (const name of manifest.skills) {
      const onDisk = existsSync(dir + '/' + name)
      const marker = onDisk ? '✓' : '✗'
      const note = onDisk ? '' : '  [missing on disk]'
      console.log(`  ${marker} ${name}  (${kindOf(name)})${note}`)
    }

    const installed = new Set(manifest.skills)
    const availableExtensions = [...extensionNames].filter((n) => !installed.has(n))
    if (availableExtensions.length > 0) {
      console.log('  available extensions (not installed):')
      for (const name of availableExtensions) {
        console.log(`    · ${name}  — install with: quaere install --skill ${name.replace(/^quaere-/, '')}`)
      }
    }
  }

  if (!found) {
    console.log('No Quaere skills installed.')
    console.log('Run: npx quaere-cli install')
  }
}
